lib/index.js

```diff
@@ -80,13 +84,43 @@ InsightAPI.prototype.getRoutePrefix = function() {
 };
 
 InsightAPI.prototype.start = function(callback) {
-  this.node.services.bitcoind.on('tx', this.transactionHandler.bind(this));
-
+  this.node.services.bitcoind.on('tx', this.transactionEventHandler.bind(this));
+  this.node.services.bitcoind.on('block', this.blockEventHandler.bind(this));
   setImmediate(callback);
 };
 
+InsightAPI.prototype.createLogInfoStream = function() {
+  var self = this;
+
+  function Log(options) {
+    Writable.call(this, options);
+  }
+  inherits(Log, Writable);
+
+  Log.prototype._write = function (chunk, enc, callback) {
+    self.node.log.info(chunk.slice(0, chunk.length - 1)); // remove new line and pass to logger
+    callback();
+  };
+  var stream = new Log();
+
+  return stream;
+};
+
 InsightAPI.prototype.setupRoutes = function(app) {
 
+  //Enable rate limiter
+  var limiter = new RateLimiter({node: this.node});
+  app.use(limiter.middleware());
+
+  //Setup logging
+  var logFormat = ':remote-addr ":method :url" :status :res[content-length] :response-time ":user-agent" ';
+  var logStream = this.createLogInfoStream();
+  app.use(morgan(logFormat, {stream: logStream}));
+
+  //Enable compression
+  app.use(compression());
```

Comment:
It's added in references to the prod2 branch: https://github.com/bitpay/insight-api/blob/prod2/config/express.js#L36 Though updated to use the latest version.